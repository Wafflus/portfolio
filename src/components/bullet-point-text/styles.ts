import styled from "styled-components";

export const BulletPointTextStyled = styled.li`
    color: ${({ theme }) => theme.data.text.lightColor};
    font-family: '${({ theme }) => theme.data.fontFamilies.outfit}', sans-serif;
    font-size: 1.125rem;
    letter-spacing: 0.1rem;
    line-height: 120%;
    list-style-type: none;
    position: relative;
    text-align: center;
    transition: transform 0.5s;

    &:hover {
        transform: scale(102%);
    }
    
    ${props => props.theme.breakpoint.above('lg',
        `
            &::before {
                background-color: ${props.theme.data.secondaryColor};
                border-radius: 50%;
                content: '';
                height: 20px;
                left: -40px;
                position: absolute;
                transition: background-color 0.5s;
                width: 20px;
            }

            &:hover::before {
                background-color: ${props.theme.data.primaryColor};
            }
        `
    )}
`;