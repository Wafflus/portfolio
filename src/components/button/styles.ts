import { IButtonStyledProps } from "./types";
import styled, { css } from "styled-components";

export const ButtonStyled = styled.a<IButtonStyledProps>`
    align-items: center;
    background-color: ${({ theme }) => theme.data.secondaryColor};
    border-radius: 8px;
    color: ${({ theme }) => theme.data.text.darkColor};
    display: flex;
    font-size: 2rem;
    font-variant: small-caps;
    gap: 16px;
    height: 66px;
    justify-content: center;
    letter-spacing: 0.1rem;
    max-width: 300px;
    min-width: 200px;
    padding: 8px;
    text-decoration: none;
    transition: transform 0.5s;
    width: 100%;

    ${props => props.outline && css`
        background-color: transparent;
        border: 1px solid ${({ theme }) => theme.data.primaryColor};
        color: #93855B;
    `}

    &:hover {
        box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.1);
        transform: scale(102%);
    }
`;