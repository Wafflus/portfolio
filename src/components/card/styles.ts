import styled from "styled-components";

export const CardStyled = styled.article`
    border-radius: 8px;
    padding: 16px;

    &:nth-child(odd) {
        background-color: ${({ theme }) => theme.data.highlightColor};
    }

    &:hover {
        outline: 3px solid ${({ theme }) => theme.data.borderColor};

        & > div > img {
            border: 3px solid ${({ theme }) => theme.data.primaryColor};
            transform: scale(102%);
        }
    }

    ${props => props.theme.breakpoint.above('lg',
        `
            &:nth-child(even) > div {
                flex-direction: row-reverse;
            }
        `
    )}
`;

export const TitleStyled = styled.h3`
    color: ${({ theme }) => theme.data.text.darkColor};
    font-size: 1.5rem;
    font-variant: small-caps;
    font-weight: 600;
    letter-spacing: 0.1rem;
    text-align: center;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            font-size: 2rem;
            text-align: start;
        `
    )}
`;

export const ToolsStyled = styled.h4`
    color: ${({ theme }) => theme.data.text.defaultColor};
    font-variant: small-caps;
    letter-spacing: 0.1rem;
    text-align: center;
`;

export const DescriptionStyled = styled.p`
    color: ${({ theme }) => theme.data.text.defaultColor};
    flex: 1;
    letter-spacing: 0.1rem;
`;