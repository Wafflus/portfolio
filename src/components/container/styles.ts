import { IContainerStyledProps } from "./types";
import styled from "styled-components";

export const ContainerStyled = styled.div<IContainerStyledProps>`
    height: 100%;
    max-width: 1440px;
    padding: ${props => props.verticalPadding} ${props => props.horizontalPadding};
    width: 100%;
    z-index: 200;
`;