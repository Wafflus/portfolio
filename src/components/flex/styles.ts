import { IFlexStyledProps } from "./types";
import styled from "styled-components";

export const FlexStyled = styled.div<IFlexStyledProps>`
    align-items: ${props => props.defaultFlexProps?.alignItems};
    display: flex;
    flex-direction: ${props => props.defaultFlexProps?.flexDirection};
    gap: ${props => props.defaultFlexProps?.gap};
    justify-content: ${props => props.defaultFlexProps?.justifyContent};
    max-width: ${props => props.defaultFlexProps?.maxWidth};
    width: 100%;
    
    ${props => props.theme.breakpoint.above('lg',
        `
            align-items: ${props.largeBreakpointFlexProps?.alignItems};
            flex-direction: ${props.largeBreakpointFlexProps?.flexDirection};
            gap: ${props.largeBreakpointFlexProps?.gap};
            justify-content: ${props.largeBreakpointFlexProps?.justifyContent};
            max-width: ${props.largeBreakpointFlexProps?.maxWidth};
        `
    )}
`;