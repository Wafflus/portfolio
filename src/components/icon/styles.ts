import styled from "styled-components";

import { IImageProps } from "./types";

export const IconStyled = styled.img<IImageProps>`
    width: ${props => props.size}${props => props.unit};
    height: ${props => props.size}${props => props.unit};
    border-radius: 50%;
`;