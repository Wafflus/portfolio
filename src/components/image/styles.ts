import styled from "styled-components";

export const ImageStyled = styled.img`
    border-radius: 8px;
    border: 3px solid transparent;
    height: 100%;
    max-height: 350px;
    max-width: 500px;
    transition: transform 0.5s, border 0.5s;
    width: 100%;
`;