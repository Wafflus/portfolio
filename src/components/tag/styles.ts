import { ITagStyledProps } from "./types";
import styled, { css } from "styled-components";

export const TagStyled = styled.a<ITagStyledProps>`
    align-items: center;
    background-color: ${({ theme }) => theme.data.secondaryColor};
    border-radius: 8px;
    color: ${({ theme }) => theme.data.text.defaultColor};
    display: flex;
    font-size: 1.5rem;
    font-variant: small-caps;
    gap: 8px;
    letter-spacing: 0.1rem;
    padding: 8px;
    text-decoration: none;
    transition: padding 0.5s;

    &:hover {
        padding: 12px;
        text-decoration: none;
    }

    ${props => !props.hasLink && css`
        cursor: default;
    `}
`;