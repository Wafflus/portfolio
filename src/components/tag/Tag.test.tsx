import "jest-styled-components";
import { render } from "@testing-library/react";
import { Tag } from ".";
import { theme } from "styles";
import { ThemeProvider } from "styled-components";

test('tag renders', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Tag href="https://www.github.com/Wafflus">
                <>Github</>
            </Tag>
        </ThemeProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toBeInTheDocument();
});

test('should have no href attribute if href is empty', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Tag href="">
                <>Github</>
            </Tag>
        </ThemeProvider>
    );

    expect(container.firstChild).not.toHaveAttribute('href');
});

test('should open in a new tab', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Tag href="https://www.github.com/Wafflus">
                <>Github</>
            </Tag>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveAttribute('target', '_blank');
});

test('should have default cursor if no link is given', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Tag href="">
                <>Github</>
            </Tag>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('cursor', 'default');
})

test('should have different padding on hover', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Tag href="https://www.github.com/Wafflus">
                <>Github</>
            </Tag>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('padding', '12px', { modifier: ':hover' });
});

test('should have no text decoration on hover', () => {
    const { container } = render(
        <ThemeProvider theme={theme}>
            <Tag href="https://www.github.com/Wafflus">
                <>Github</>
            </Tag>
        </ThemeProvider>
    );

    expect(container.firstChild).toHaveStyleRule('text-decoration', 'none', { modifier: ':hover' });
});