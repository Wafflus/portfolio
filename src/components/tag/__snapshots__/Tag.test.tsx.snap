// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tag renders 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFECB1;
  border-radius: 8px;
  color: #93855B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.5rem;
  font-variant: small-caps;
  gap: 8px;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  padding: 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: padding 0.5s;
  transition: padding 0.5s;
}

.c0:hover {
  padding: 12px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
  class="c0"
  href="https://www.github.com/Wafflus"
  target="_blank"
>
  Github
</a>
`;
